test/k8s-integration/main.go

```diff
@@ -0,0 +1,341 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"syscall"
+
+	"github.com/golang/glog"
+
+	testutils "sigs.k8s.io/gcp-compute-persistent-disk-csi-driver/test/e2e/utils"
+
+	"k8s.io/apimachinery/pkg/util/uuid"
+)
+
+var (
+	teardownCluster = flag.Bool("teardown-cluster", true, "teardown the cluster after the e2e test")
+	stagingImage    = flag.String("staging-image", "gcr.io/dyzz-csi-staging/csi/gce-pd-driver", "name of image to stage to")
```

Comment:
What is the container registry when run in prow?

Reply:
the project we get from boskos loaning. The test should do its own GC as well, and we plan on adding janitor logic to clean up orphaned images too: https://github.com/kubernetes/test-infra/issues/9597

Reply:
do you want to encode your project name into the binary? what if it's just a required argument?